### PR TITLE
Update DCT docs with 1.11 info, fix typos

### PR DESCRIPTION
**Pull request by riyazdf**
Revamps some of the Docker Content Trust docs with more up-to-date information with respect to hardware signing and registry compatibility.  Also fixes some typos.

cc @docker/security-team 

Signed-off-by: Riyaz Faizullabhoy riyaz.faizullabhoy@docker.com

### CODE REVIEW

File: docs/security/trust/deploying_notary.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ The easiest way to deploy Notary Server is by using Docker Compose. To follow th
         docker-compose up -d
 
 
-    For more detailed documentation about how to deploy Notary Server see https://github.com/docker/notary.
+    For more detailed documentation about how to deploy Notary Server see the [instructions to run a Notary service](https://github.com/docker/notary/blob/master/docs/running_a_service.md) as well as https://github.com/docker/notary for more information.
```
Comment (cyli) [inline docs/security/trust/deploying_notary.md]: Should we generally be linking to https://docs.docker.com/notary here instead of github?

File: docs/security/trust/content_trust.md [hunk 2]
```diff
@@ -14,14 +14,10 @@ weight=-1
 When transferring data among networked systems, *trust* is a central concern. In
 particular, when communicating over an untrusted medium such as the internet, it
 is critical to ensure the integrity and publisher of all the data a system
-operates on. You use Docker to push and pull images (data) to a registry. Content trust
+operates on. You use Docker to push and pull images (data) to a public or private registry. Content trust
 gives you the ability to both verify the integrity and the publisher of all the
```
Comment (cyli) [inline docs/security/trust/content_trust.md]: I know this wasn't part of your change, but the next sentence is phrased a little oddly.  Maybe "verify both the integrity and the publisher"...?

File: docs/security/trust/deploying_notary.md (hunk 1 above)
Comment (riyazdf) [inline docs/security/trust/deploying_notary.md]: Ah yes, agreed!

File: docs/security/trust/content_trust.md (hunk 2 above)
Comment (riyazdf) [inline docs/security/trust/content_trust.md]: good point, I'll update this :)

**Comment by cyli** [comment]
Thanks @riyazdf!  LGTM

File: docs/security/trust/deploying_notary.md [hunk 3]
```diff
@@ -21,7 +21,7 @@ The easiest way to deploy Notary Server is by using Docker Compose. To follow th
         docker-compose up -d
 
 
-    For more detailed documentation about how to deploy Notary Server see https://github.com/docker/notary.
+    For more detailed documentation about how to deploy Notary Server see the [instructions to run a Notary service](https://docs.docker.com/notary/running_a_service/) as well as https://github.com/docker/notary for more information.
```
Comment (SvenDowideit) [inline docs/security/trust/deploying_notary.md]: this should link to the markdown file - `/notary/running_a_service.md`, so that versioned linking works.

File: docs/security/trust/trust_key_mng.md [hunk 4]
```diff
@@ -64,6 +64,16 @@ Before backing them up, you should `tar` them into an archive:
 $ umask 077; tar -zcvf private_keys_backup.tar.gz ~/.docker/trust/private; umask 022
 ```
 
+## Hardware storage and signing
+
+Docker Content Trust can store and sign with root keys from a Yubikey 4.
+The Yubikey will be prioritized over keys stored in the filesystem; if a root key
+is not found locally when initializing a new repository with content trust, Docker
+Engine will prioritize creating a root key in a Yubikey if it can find one.
+Please consult the [Notary documentation](https://docs.docker.com/notary/advanced_usage/#use-a-yubikey) for more details.
```
Comment (SvenDowideit) [inline docs/security/trust/trust_key_mng.md]: should be `/notary/advanced_usage.md#use-a-yubikey`

File: docs/security/trust/deploying_notary.md (hunk 3 above)
Comment (riyazdf) [inline docs/security/trust/deploying_notary.md]: thanks, updated!

File: docs/security/trust/trust_key_mng.md (hunk 4 above)
Comment (riyazdf) [inline docs/security/trust/trust_key_mng.md]: thanks, done!  I've also updated other notary documentation links to be consistent

File: docs/security/trust/content_trust.md [hunk 5]
```diff
@@ -13,15 +13,11 @@ weight=-1
 
 When transferring data among networked systems, *trust* is a central concern. In
 particular, when communicating over an untrusted medium such as the internet, it
-is critical to ensure the integrity and publisher of all the data a system
-operates on. You use Docker to push and pull images (data) to a registry. Content trust
-gives you the ability to both verify the integrity and the publisher of all the
+is critical to ensure the integrity and the publisher of all the data a system
+operates on. You use Docker to push and pull images (data) to a public or private registry. Content trust
```
Comment (moxiegirl) [inline docs/security/trust/content_trust.md]: Docker > Docker Engine  (we need to state clearly which product does what)

Comment (riyazdf) [inline docs/security/trust/content_trust.md]: Got it, I've updated the usage here.

File: docs/security/trust/trust_key_mng.md [hunk 6]
```diff
@@ -64,6 +64,16 @@ Before backing them up, you should `tar` them into an archive:
 $ umask 077; tar -zcvf private_keys_backup.tar.gz ~/.docker/trust/private; umask 022
 ```
 
+## Hardware storage and signing
+
+Docker Content Trust can store and sign with root keys from a Yubikey 4.
```
Comment (moxiegirl) [inline docs/security/trust/trust_key_mng.md]: There are about four ideas in this and they are all smushed together.  Hard to parse. Try:

```markdown
Docker Content Trust can store and sign with root keys from a Yubikey 4. The
Yubikey is prioritized over keys stored in the filesystem. When you initialize a
new repository with content trust, the Engine looks for a root key locally. If a
key is not found and the Yubikey 4 exists, the Engine creates a root key in
Yubikey 4.
```

Comment (riyazdf) [inline docs/security/trust/trust_key_mng.md]: Thanks, I've worked in your suggestion 👍

**Comment by moxiegirl** [comment]
LGTM Thanks @riyazdf
